pkg/k8sutil/create_test.go

```diff
@@ -287,3 +288,177 @@ func TestCreateOrUpdateNamespaceUpdateSuccess(t *testing.T) {
 		t.Fatalf("expected %q, got: %q", name, mockns.ObjectMeta.Labels[internal.NamespaceLabelKey])
 	}
 }
+
+func TestYAMLToUnstructured(t *testing.T) {
+	tests := []struct {
+		name    string
+		yamlObj []byte
+		want    *unstructured.Unstructured
+		wantErr bool
+	}{
+		{
+			name: "Valid_Kubernetes_object",
+			yamlObj: []byte(`apiVersion: v1
+kind: Namespace
+metadata:
+  name: test`),
+			want: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "Namespace",
+					"metadata":   map[string]interface{}{"name": "test"},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "Invalid_Kubernetes_object",
+			yamlObj: []byte(`foobar`),
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Perhaps we can call `t.Parallel()` in here. There is a new linter which checks if all tests are parallelized.